### ISSUE TITLE
Pure-Text Target: When generating the HTML for text output, allow the stylesheets to optimize for text somehow

### ISSUE THREAD

**Issue by ghost** [openSUSE/daps#312]
DAPS generates pure-text output by first generating HTML and then taking a "screenshot" of the page rendered in `w3m`. This is the only option that works well enough for, e.g., keeping tables intact.

At the moment, DAPS generates the HTML version with the defualt options for the stylesheets by default. However, that is not always helpful. E.g., with the `suse2013` stylesheets, by default, the HTML output includes  `#`s that are used as permalinks. In the text output, however, these are simple `#`s that stand around uselessly, cluttering the landscape.

Possible solutions for this include:
1.  DAPS uses a non-configurable list of xsltparams which is geared toward the most-often used stylesheets (upstream, suse2005, suse2013) to generate its `xsltproc`/`saxon` call with.
   
   **Caveats:** This not being configurable means that it may not be able to work properly on new/unknown stylesheets.
2.  DAPS uses a configurable list of xsltparams which, by default, is geared toward the most-often used stylesheets  (upstream, suse2005, suse2013) to generate its `xsltproc`/`saxon` call with. Basically, like the `XSLPARAM` option, but only for text output.
   
   **Caveats:** By default, this may produce unexpected results with new/unknown stylesheets.
3.  DAPS gives the stylesheets a single parameter, e.g. `final.output.text=1` that allows the stylesheets to react.
   
   **Caveats:** Would not be supported by the upstream stylesheets.

**Comment by tomschr** [comment]
> However, that is not always helpful. E.g., with the suse2013 stylesheets [...]

Text is **not** generated through the suse2013 stylesheets. By default, daps uses the upstream XHTML stylesheets. I used it to have output that is mostly "clean" and independant from our modifications.

For example, if you run daps with the GNOME User Guide in debug mode, you will see this:

```
$ daps -vvv -d DC-SLED-gnomeuser text
[... output pruned ...]
/local/doc/daps/libexec/daps-xslt --stringparam "rootid=book.gnomeuser"    \
  --output /local/doc/GH/doc-sle/build/.tmp/book.gnomeuser.html --xinclude \
  --stylesheet /usr/share/xml/docbook/stylesheet/nwalsh5/1.78.1/xhtml/docbook.xsl \
  --file /local/doc/GH/doc-sle/build/.profiled/x86_64_zseries_power_sled/MAIN.SLEDS.xml \
  --param "show.comments=0"  \
  /usr/bin/xsltproc
```

As you can see, the upstream stylesheets from `/usr/share/xml/docbook/stylesheet/nwalsh5/1.78.1/xhtml/docbook.xsl` are used.

Maybe I miss something, but I don't see an issue here...

**Comment by ghost** [comment]
> Maybe I miss something, but I don't see an issue here...

I think you miss something here. ;)

Text can be generated through any kind of stylesheet. Incidentally, you are right, current DAPS versions use suse2005 _by default_. Now, the issue is that the suse2005 stylesheets are less actively developed and may go away at some point.

In any case, using `--styleroot`, you can change the stylesheets for text generation.

Historically, the default for text generation changed from upstream => suse2013 => suse2005. The version of DAPS shipped with SLE 12 SP1 actually uses the suse2013 stylesheets. (Which we noticed when Ihno stormed into our room yesterday, annoyed that there were hashes at the end of every headline in the SLE 12 SP1 GA documentation.)

**Comment by fsundermeyer** [comment]
sknorr. tomschr: Please see https://github.com/openSUSE/daps/commit/005e214f

Anyway, this is not about the stylesheets being used, but rather about providing parameters which guarantee a useful text output.

**Comment by ghost** [comment]
True. That was the point I should have tried to make. ;)

**Comment by fsundermeyer** [comment]
Fixed with commit a146446d6
- parameters for text generation can be set in the config files via
  TXT_PARAMS or via --param/--stringparam on the command line
- a change introduced in 2.0 made using the original DocBook stylesheets
  for text generation the default. Since this behavior is confusing and
  unexpected, it is reverted with this change:
  - the stylesheets configured via STYLEROOT or --styleroot are used by
    default for text generation
  - to ignore _any_ STYLEROOT definitions, set
    TXT_IGNORE_STYLEROOT="yes" in the config files or use
    --ignore-styleroot on the command line
    (daps -d <DC> text --ignore-styleroot)
